runtime/queries/metricsview.go

```diff
@@ -258,3 +258,52 @@ func repeatString(val string, n int) []string {
 	}
 	return res
 }
+
+func validateAndGetDimension(mv *runtimev1.MetricsView, dimName string) (string, error) {
+	dimName = strings.ToLower(dimName)
+	for _, dimension := range mv.Dimensions {
+		if strings.EqualFold(dimension.Name, dimName) {
+			if dimension.Column != "" {
+				return dimension.Column, nil
+			}
+			// backwards compatibility for older projects that have not run reconcile on this dashboard
+			// in that case `column` will not be present
+			return dimension.Name, nil
+		}
+	}
+	return "", fmt.Errorf("dimension %s not found", dimName)
+}
+
+func convertFilterToColumn(mv *runtimev1.MetricsView, filter *runtimev1.MetricsViewFilter) error {
```

Comment:
Isn't this the only place that we lookup dimension columns in a filter? https://github.com/rilldata/rill-developer/blob/2efa20aa0de5694190029c68aeadf8a19be2a197/runtime/queries/metricsview.go#LL175C18-L175C18

I think it should just call `validateAndGetDimension` right there instead of doing this in-place rewriting